### ISSUE TITLE
How to properly use custom adapter?

### ISSUE THREAD

**Issue by nightire** [jasonmit/ember-intl#409]
I'm trying to use a custom adapter in one of my add-on by extend and overwrite `IntlService.adapter` CP, it seems work but still execute the default logic before I actually hit the custom adapter, see this screenshot:

![image](https://cloud.githubusercontent.com/assets/1104018/21542986/1804c240-cdfd-11e6-9439-64a97fb216f7.png)

This is how I do it for now:

1. in add-on, create `/addon/adapters/default.js`, it is extended from `ember-intl/adapters/default`
2. in add-on, create `/addon/services/intl.js`, it is extended from `ember-intl/services/intl`:
    ```js
    export default IntlService.extend({
      adapter: computed({
        get() {
          return getOwner(this).lookup('ember-choice-i18n@adapter:default')
        }
      })
    })
    ```

I don't know what else need to make custom adapter works without error in screenshot above, any help would be appreciated.

**Comment by jasonmit** [comment]
Any chance you can isolate this and push it to github somewhere so I can step through it?

**Comment by nightire** [comment]
@jasonmit I can't reproduce the error again, seems like custom adapter has already works as I've expected, don't know why the error happened before.

**Comment by jasonmit** [comment]
Great to hear!  I'm going to go ahead and close this.

**Comment by nightire** [comment]
@jasonmit But I also observe another problem on https://github.com/jasonmit/ember-intl/blob/master/packages/ember-intl/app/utils/intl/missing-message.js#L13

sometimes the passing `locales` is not an array, so `locales.join is not a function` error will be thrown.

for example, if simply put: `{{t "key.not.exists" locale="en-us"}}` in template will reproduce this error

**Comment by jasonmit** [comment]
@nightire thanks, I just released 2.16.3 to fix that

**Comment by nightire** [comment]
@jasonmit awesome, thank you very much.
